### PR TITLE
Initial template to deploy Azure Stack HCI 23H2 via ARM Template

### PR DESCRIPTION
**Pull request by troettinger**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*Adding the initial version of the JSON ARM template
*
*

### CODE REVIEW

**Comment by troettinger** [comment]
The template was validated manually using our CI lab.

File: quickstarts/microsoft.azurestackhci/create-cluster/metadata.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "creates an Azure Stack HCI 23H2 cluster",
+  "description": "This template creates an Azure Stack HCI 23H2 cluster using an ARM template.",
+  "summary": "This template creates an Azure Stack HCI 23H2 cluster using an ARM template.",
+  "githubUsername": "troettinger",
+  "validationType": "Manual",
```
Comment (alex-frankel) [inline quickstarts/microsoft.azurestackhci/create-cluster/metadata.json:8]: Why is manual validation required?

**Review by alex-frankel — COMMENTED** [review]
Need to better understand why manual approval is needed for this case

File: quickstarts/microsoft.azurestackhci/create-cluster/metadata.json (hunk 1 above)
Comment (troettinger) [reply quickstarts/microsoft.azurestackhci/create-cluster/metadata.json:8]: Validation can be automated for sure but requires hardware wired correctly as the template does deploy a Windows Failover cluster using hardware. Hence i did pick manual validation as its not deploying a bunch of VMs.

**Pull request merged**